module/zfs/abd.c

```diff
@@ -366,6 +366,32 @@ abd_alloc_gang_abd(void)
 	return (abd);
 }
 
+/*
+ * If we are adding a gang ABD to another gang ABD we will iterate
+ * though the child gang's individual ABD and add them individually
+ * to the parent.
+ */
+static void
+abd_gang_add_gang(abd_t *pabd, abd_t *cabd)
+{
+	abd_t *child = NULL;
+
+	ASSERT(abd_is_gang(pabd));
+	ASSERT(abd_is_gang(cabd));
+
+	for (child = list_head(&ABD_GANG(cabd).abd_gang_chain);
+	    child != NULL;
+	    child = list_next(&ABD_GANG(cabd).abd_gang_chain, child)) {
+		/*
+		 * We always pass B_FALSE for free_on_free as it is the
+		 * original child gang ABDs responsibilty to determine
+		 * if the any of it's children ABD should be free'd on
```

Comment:
nit: "if the any of it's children ABD should be free'd" -> "if any of its child ABDs should be free'd"